app/src/main/java/com/simply/birthdayapp/presentation/ui/screens/auth/signIn/forgotpassword/ForgotPasswordScreen.kt

```diff
@@ -0,0 +1,204 @@
+package com.simply.birthdayapp.presentation.ui.screens.auth.signIn.forgotpassword
+
+
+import android.widget.Toast
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.layout.wrapContentHeight
+import androidx.compose.foundation.rememberScrollState
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.foundation.verticalScroll
+import androidx.compose.material3.AlertDialog
+import androidx.compose.material3.Card
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.text.input.ImeAction
+import androidx.compose.ui.text.input.KeyboardType
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.simply.birthdayapp.R
+import com.simply.birthdayapp.presentation.ui.components.AppBaseTopBar
+import com.simply.birthdayapp.presentation.ui.components.AuthButton
+import com.simply.birthdayapp.presentation.ui.components.BaseTextField
+import com.simply.birthdayapp.presentation.ui.theme.AppTheme
+import org.koin.androidx.compose.getViewModel
+
+@Composable
+fun ForgotPasswordScreen(
+    forgotPasswordViewModel: ForgotPasswordViewModel,
+    onBackClick: () -> Unit = {},
+    onNewPasswordButtonClick: () -> Unit = {}
+) {
+    val getCodeErrorMessage by forgotPasswordViewModel.getCodeErrorMessage.collectAsState()
+    val hasEmailError by forgotPasswordViewModel.hasEmailError.collectAsState()
+    val email by forgotPasswordViewModel.email.collectAsState()
+    val code by forgotPasswordViewModel.code.collectAsState()
+    val showPasswordCodeSection by forgotPasswordViewModel.hasGetCodeSuccess.collectAsState()
+    val hasGetCodeSuccess by forgotPasswordViewModel.hasGetCodeSuccess.collectAsState()
+    val getCodeErrorState by forgotPasswordViewModel.getCodeErrorState.collectAsState()
+    val emailMaxLength = 30
+    val codeMaxLength = 6
+    val enabledGetCodeButton: Boolean = email.isNotEmpty() && !hasEmailError
```

Comment:
This check may be inside viewModel. We can have a stateflow that will combine these 2 flows and do this check